features/api_demo/test_countries_rest_api.py

```diff
@@ -0,0 +1,118 @@
+    
+    un_members = []
+    for country in range(len(response)):
+        if response[country]["unMember"]:
+            un_members.append(response[country]["name"]["common"])
+        else:
+            pass
+    
+    when(Arlong).attempts_to(
+        See.the(un_members, ContainsTheItem("United States"))
+    )
+
+# use parametrize to pit one test function against multiple inputs and assertions
+@pytest.mark.parametrize("country_name,codes,symbols", countries)
+def test_currency_codes(Arlong: Actor, country_name, codes, symbols) -> None:
+    """ensure each country's currency matches the code on file"""
+    given(Arlong).was_able_to(
+        SendGETRequest.to(f"{BY_NAME}/{country_name}{FULL_NAME}"),
+    )
+    with the_narrator.off_the_air():
+        response = BodyOfTheLastResponse().answered_by(Arlong).pop()
+    when(Arlong).attempts_to(
+        Pause.for_(1).second_because("we don't want to spam the API.")
+    )
+    currency_keys = list(response["currencies"].keys())
+    currency_values = response["currencies"].values()
+    response_symbols = []
+    for key in (currency_values):
+        currency_symbol = key["symbol"]
+        response_symbols.append(currency_symbol)
+    
+
+    then(Arlong).should(
+        See.the(currency_keys, IsEqualTo(codes)),
+        See.the(response_symbols, IsEqualTo(symbols))
+    )
+
+# actually, building the params for the prior test was a much bigger flex
+def test_build_currency_param_list(Arlong: Actor) -> None:
+    """We go way out of our way here to do something stupidly complex. The point is to
+    deal with a lot of bullshit traversing in a big nasty dict."""
+    given(Arlong).was_able_to(SendGETRequest.to(ALL_COUNTRIES))
+    with the_narrator.off_the_air():
+        response = BodyOfTheLastResponse().answered_by(Arlong)
+    
+    country_names = ['Tuvalu', 'Lebanon', 'Burkina Faso', 'British Virgin Islands', 'El Salvador', 'Timor-Leste', 'Italy', 'Palestine', 'Guam', 'Faroe Islands', 'Afghanistan', 'São Tomé and Príncipe', 'China', 'Panama', 'Venezuela', 'Micronesia', 'French Southern and Antarctic Lands', 'Finland', 'Colombia', 'Bhutan', 'Jersey', 'Mexico', 'Denmark', 'Isle of Man']
+
+    fuckers_trick = []
+    for item in range(len(response)):
+        current_name = response[item]["name"]["common"]
+        if  current_name in country_names: 
+            currencies_list = []
+            symbols_list = []
+            try:
+                codes = list(response[item]["currencies"].keys())
+                has_currency = response[item]["currencies"].values()
+                list_len = 0
+                for key in (has_currency):
+                    code_name = codes[list_len]
+                    currencies_list.append(code_name)
+                    currency_symbol = key["symbol"]
+                    symbols_list.append(currency_symbol)
+                    list_len +=1
+                fuckers_trick.append((current_name, currencies_list, symbols_list))
```

Comment:
The motivation for this test file was me absolutely shitting the bed during a recent code interview. It had been years since I had bothered doing any serious traversal of complex nested dict/list structures. I struggled brutally with simple issues including how to work back to evaluating a simple assertion using my own familiar framework. Needless to say, hindsight is too late, but at least I won't be caught off guard by having to deal with a monster JSON payload ever again.